### PR TITLE
Fix/ch26280/do not show all collections in conditions

### PR DESCRIPTION
**Pull request by ericreis**
#### What is the purpose of this pull request?
Do not show all collection in conditions statements.

#### What problem is this solving?
There are stores that have many collections (C&A has 8k). To query for those collections We were using a route that was not searchable nor paginated. This request would always timeout for those stores.

### CODE REVIEW

File: react/components/Promotion/Conditions/Objects/ProductsSelectObject.tsx [hunk 1]
```diff
@@ -2,19 +2,19 @@ import React from 'react'
...
 import { mapProductsToSelect } from '../../../../utils/mappers'
 
 export interface Props extends SelectObjectProps, WithProductsProps {
-  productsData: ProductsData['getProducts']
+  products: Product[]
 }
 
 const ProductsSelectObject: React.FC<Props & InjectedIntlProps> = ({
-  productsData,
+  products = [],
```
Comment (emersonlaurentino) [inline react/components/Promotion/Conditions/Objects/ProductsSelectObject.tsx]: why don't u use `ProductsSelectObject.defaultProps`?

File: react/connectors/withCollections.tsx [hunk 2]
```diff
@@ -23,17 +22,21 @@ const withCollections = <T extends WithCollectionsProps>(
...
           <WrappedComponent
             {...props}
             loading={loading}
             error={error && error.message}
-            collections={data && data.getCollections}
-            searchForCollections={searchTerm => setName(searchTerm)}
+            collections={
+              data && data.getCollections && data.getCollections.collections
+            }
+            searchForCollections={searchTerm => setSearchTerm(searchTerm)}
```
Comment (emersonlaurentino) [inline react/connectors/withCollections.tsx]:
```suggestion
            searchForCollections={setSearchTerm}
```

File: react/connectors/withProducts.tsx [hunk 3]
```diff
@@ -34,7 +31,7 @@ const withProducts = <T extends WithProductsProps>(
             {...props}
             loading={loading}
             error={error && error.message}
-            productsData={data && data.getProducts}
+            products={data && data.getProducts && data.getProducts.products}
             searchForProducts={searchTerm => setName(searchTerm)}
```
Comment (emersonlaurentino) [inline react/connectors/withProducts.tsx]:
```suggestion
            searchForProducts={setName}
```

File: react/components/Promotion/Conditions/Objects/ProductsSelectObject.tsx (hunk 1 above)
Comment (ericreis) [reply react/components/Promotion/Conditions/Objects/ProductsSelectObject.tsx]: I dislike using `.defaultProps` when using TS FCs. I think this is more readable.

Comment (emersonlaurentino) [reply react/components/Promotion/Conditions/Objects/ProductsSelectObject.tsx]: :disappointed_relieved:

**Pull request merged**